vendor/engines/secure_rooms/app/controllers/secure_rooms/facility_occupancies_controller.rb

```diff
@@ -2,19 +2,73 @@ module SecureRooms
 
   class FacilityOccupanciesController < ApplicationController
 
+    include ProblemOrderDetailsController
     include TabCountHelper
 
     admin_tab     :all
     before_action :authenticate_user!
     before_action :check_acting_as
     before_action :init_current_facility
+    before_action :sanitize_sort_params, only: :index
+
+    load_and_authorize_resource class: Occupancy
+
+    SORTING_CLAUSES = {
+      "entry_at" => "secure_rooms_occupancies.entry_at",
+      "user_name" => "users.last_name",
```

Comment:
Let's stick with what the other pages are doing. I think there are usually few enough orders on these views that the indexing doesn't really matter.